File: model/Vaccines.go
package model

import (
	"time"
	"github.com/jinzhu/gorm"
)

type Vaccines struct {
	gorm.Model `swaggerignore:"true"`
	AnimalID   int    `json:"animalID"`
	VacineName string `json:"vaccineName"`
	Date       time.Time `json:"date"`							//Data para tomar a vacina 
	Taken 	   bool `json:"taken"`								//Se tomou ou não										
	DateTaken  time.Time `json:"dateTaken"`						//Data em que tomou a vacina
	Validity   time.Time `json:"validity"`						//Data de validade da vacina (aproximado)
}

File: src/app/models/vaccines.ts
export class Vaccines {
    date: Date;
    taken: boolean;
    dateTaken:Date;
    validity:Date;
    vaccineName:String;
    AnimalID:Number;

    constructor(animalID:number,date:Date,taken:boolean,validity:Date,vaccineName:String,dateTaken:Date) {
        this.AnimalID = animalID;
        this.date = date;
        this.taken = taken;
        this.validity = validity;
        this.vaccineName = vaccineName;
        this.dateTaken = dateTaken
    }
}
